server/api/src/rsDataObjChksum.cpp

```diff
@@ -34,34 +38,22 @@ namespace {
         }
         const rodsLong_t size_in_vault = fileStatOut->st_size;
         free(fileStatOut);
-        genQueryInp_t genQueryInp;
-        memset(&genQueryInp, 0, sizeof(genQueryInp));
-        addInxIval(&genQueryInp.selectInp, COL_DATA_SIZE, 1);
-        addInxVal(&genQueryInp.sqlCondInp, COL_D_DATA_PATH, (boost::format(" = '%s'") % dataObjInfo->filePath).str().c_str());
-        genQueryInp.maxRows = 1;
-        genQueryOut_t *genQueryOut = nullptr;
-        const int status_rsGenQuery = rsGenQuery(rsComm, &genQueryInp, &genQueryOut);
-        clearGenQueryInp(&genQueryInp);
-        if (status_rsGenQuery < 0) {
-            if (status_rsGenQuery == CAT_NO_ROWS_FOUND) {
-                freeGenQueryOut(&genQueryOut);
+        std::optional<rodsLong_t> size_in_database {};
+        std::string select_and_condition {
+            (boost::format("select DATA_SIZE where DATA_PATH = '%s'") % dataObjInfo->filePath).str()
```

Comment:
There is nothing wrong with this code, but use the new `fmt` library instead of `boost::format`. The `fmt` library is significantly faster. With that, this line could be this:
```c++
const auto gql = fmt::format("select DATA_SIZE where DATA_PATH = {}", dataObjInfo->filePath);
```
This library uses Python-like format specifiers, and it automatically returns a `std::string` when called so you can use `auto` to decrease typing and keep the code tighter.